Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -305,144 +273,140 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
 
             // Assert
-            executedTask.RuleSetFilePath.Should().Be("c:\\test.project.ruleset.vb");
-            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo("c:\\sonar.vbnet1.dll", "c:\\Google.Protobuf.dll");
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("c:\\add1.txt.vb", "d:\\replaced1.txt");
+            executedTask.RuleSetFilePath.Should().Be(expectedRuleset);
+            executedTask.AnalyzerFilePaths.Should().BeEquivalentTo($@"c:\sonar.{language}.dll", @"c:\Google.Protobuf.dll");
+            executedTask.AdditionalFilePaths.Should().BeEquivalentTo($@"c:\add1.{language}.txt", @"d:\replaced1.txt");
         }
 
         [TestMethod]
         public void GetAnalyzerSettings_ConfigExists_NewBehaviour_TestProject_SonarAnalyzerSettingsUsed_UnknownLanguage()
         {
             // Arrange and Act
-            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.4", "unknownLang");
+            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.4", "unknownLang", true, null);
 
             // Assert
             executedTask.RuleSetFilePath.Should().BeNull();
             executedTask.AnalyzerFilePaths.Should().BeNull();
-            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("original.should.be.removed.txt", "original.should.be.replaced\\replaced1.txt");
+            executedTask.AdditionalFilePaths.Should().BeEquivalentTo("original.should.be.removed.for.excluded.test.txt", "original.should.be.replaced\\replaced1.txt");
         }
 
-        private GetAnalyzerSettings ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language)
+        private GetAnalyzerSettings ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language, bool isTestProject, string excludeTestProject)
         {
             // Want to test the behaviour with old and new SQ version. Expecting the same results in each case.
-
             // Arrange
             var config = new AnalysisConfig
             {
                 SonarQubeVersion = sonarQubeVersion,
                 ServerSettings = new AnalysisProperties
                 {
-                    // Setting should be ignored
-                    new Property { Id = "sonar.cs.roslyn.ignoreIssues", Value = "true" }
+                    // Server settings should be ignored
+                    new Property { Id = "sonar.cs.roslyn.ignoreIssues", Value = "true" },
+                    new Property { Id = "sonar.vbnet.roslyn.ignoreIssues", Value = "true" },
+                    // Server settings should be ignored - it should never come from the server
+                    new Property { Id = "sonar.dotnet.excludeTestProjects", Value = "true" }
```

Comment:
if it should never come from the server, could you give it a faulty value, like "foo" or "42", rather than a boolean one? I guess this applies for all "should be ignored" server settings

Reply:
I need `true` here, because it would break at least one UT if the implementation would read it.
Having `foo` is the same as the test case with `unexpectedValue` and that behaves the same as `false` => nothing would break if the implementation changes.

Same logic probably applies to `ignoreIssues` as well.

Reply:
Ok, maybe a small comment saying that "we use true because it would break tests if it were read" would help for these should-be-ignored values